system-addon/lib/UserDomainAffinityProvider.jsm

```diff
@@ -4,8 +4,9 @@
 "use strict";
 
 ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
 
-const history = Cc["@mozilla.org/browser/nav-history-service;1"].getService(Ci.nsINavHistoryService);
+XPCOMUtils.defineLazyServiceGetter(this, "NavHistoryService", "@mozilla.org/browser/nav-history-service;1", "nsINavHistoryService");
```

Comment:
Oh, alternatively `ChromeUtils.defineModuleGetter(this, "PlacesUtils", "resource://gre/modules/PlacesUtils.jsm");` then access `PlacesUtils.history`?